pkg/prometheus/promcfg.go

```diff
@@ -695,16 +712,7 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 		}
 	}
 
-	if ep.BasicAuth != nil {
-		if s, ok := store.BasicAuthAssets[fmt.Sprintf("podMonitor/%s/%s/%d", m.Namespace, m.Name, i)]; ok {
-			cfg = append(cfg, yaml.MapItem{
-				Key: "basic_auth", Value: yaml.MapSlice{
-					{Key: "username", Value: s.Username},
-					{Key: "password", Value: s.Password},
-				},
-			})
-		}
-	}
+	cfg = cg.WithMinimumVersion("2.26.0").addBasicAuthToYaml(cfg, fmt.Sprintf("podMonitor/%s/%s/%d", m.Namespace, m.Name, i), store, ep.BasicAuth)
```

Comment:
there's no condition on the Prometheus version for pod scrape configuration.
```suggestion
	cfg = cg.addBasicAuthToYaml(cfg, fmt.Sprintf("podMonitor/%s/%s/%d", m.Namespace, m.Name, i), store, ep.BasicAuth)
```